tests/e2e/requires_aws_api.go

```diff
@@ -0,0 +1,347 @@
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/driver"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/testsuites"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	volumesnapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+	. "github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclientset "k8s.io/client-go/rest"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	awscloud "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/cloud"
+	ebscsidriver "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver"
+)
+
+const testTagName = "testTag"
+const testTagValue = "3.1415926"
+
+// Tests that require that the e2e runner has access to the AWS API
+var _ = Describe("[ebs-csi-e2e] [single-az] [requires-aws-api] Dynamic Provisioning", func() {
+	f := framework.NewDefaultFramework("ebs")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	var (
+		cs          clientset.Interface
+		snapshotrcs restclientset.Interface
+		ns          *v1.Namespace
+		ebsDriver   driver.PVTestDriver
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		var err error
+		snapshotrcs, err = restClient(testsuites.SnapshotAPIGroup, testsuites.APIVersionv1)
+		if err != nil {
+			Fail(fmt.Sprintf("could not get rest clientset: %v", err))
+		}
+		ns = f.Namespace
+		ebsDriver = driver.InitEbsCSIDriver()
+	})
+
+	ec2Client := ec2.New(session.Must(session.NewSession()))
+
+	It("should create a snapshot with FSR enabled", func() {
+		azList, err := ec2Client.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+		if err != nil {
+			Fail(fmt.Sprintf("failed to list AZs: %v", err))
+		}
+		fsrAvailabilityZone := *azList.AvailabilityZones[0].ZoneName
+
+		pod := testsuites.PodDetails{
+			// sync before taking a snapshot so that any cached data is written to the EBS volume
+			Cmd: "echo 'hello world' >> /mnt/test-1/data && grep 'hello world' /mnt/test-1/data && sync",
+			Volumes: []testsuites.VolumeDetails{
+				{
+					CreateVolumeParameters: map[string]string{
+						ebscsidriver.VolumeTypeKey: awscloud.VolumeTypeGP3,
+						ebscsidriver.FSTypeKey:     ebscsidriver.FSTypeExt4,
+					},
+					ClaimSize: driver.MinimumSizeForVolumeType(awscloud.VolumeTypeGP3),
+					VolumeMount: testsuites.VolumeMountDetails{
+						NameGenerate:      "test-volume-",
```

Comment:
Let's rely on [DefaultVolumeName](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/61c721b88933faec8fbb9c6610d5fc9d265ef5d2/tests/e2e/testsuites/e2e_utils.go#L20C2-L20C2) and [DefaultMountPath](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/61c721b88933faec8fbb9c6610d5fc9d265ef5d2/tests/e2e/testsuites/e2e_utils.go#L21) for these test's VolumeMount's? (Or create a VolumeMount const in e2e_utils shared across each test-case?)

(To not have hanging constants. Yes, the old dynamic provisioning tests would need to be refactored to use these as well, but I think that's a PR worth making.)

Reply:
I don't using those defaults is reasonable because this is a generator so those values don't make sense. I created a `DefaultVolumeMountGenerator` const instead.